File: src/test/testData/java/implicit/HashTable.java
package ru.ustinov;

import java.net.URL;
import java.util.Hashtable;

class Simple {
    public static void main() {
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">Hashtable<URL, Integer> a;</warning>
        Hashtable<Integer, URL> b;
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">Hashtable<Hashtable<Integer, URL>, Integer> c, d;</warning>
        Hashtable<Integer, Hashtable<Integer, URL>> e, f;
    }
}

File: src/test/testData/kotlin/implicit/HashTable.kt
import java.net.URL
import java.util.Hashtable
import java.util.HashSet

@Suppress("UNUSED_VARIABLE")
fun main() {
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val a: Hashtable<URL, Int></warning>
    val b: Hashtable<Int, URL>
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val c: Hashtable<HashSet<URL>, Int></warning>
    val d: Hashtable<Int, HashSet<Int>>
}